extensions/banner/runtime/src/main/java/io/quarkus/banner/runtime/BannerRecorder.java

```diff
@@ -0,0 +1,13 @@
+package io.quarkus.banner.runtime;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.runtime.annotations.Recorder;
+
+@Recorder
+public class BannerRecorder {
+    public void print(String banner) {
+        Logger logger = Logger.getLogger("io.quarkus");
+        logger.info(banner);
```

Comment:
The banner shouldn't be logged, it should be [implemented as a header](https://docs.oracle.com/en/java/javase/11/docs/api/java.logging/java/util/logging/Formatter.html#getHead(java.util.logging.Handler)) within [the console setup code](https://github.com/quarkusio/quarkus/blob/master/core/runtime/src/main/java/io/quarkus/runtime/logging/LoggingSetupRecorder.java#L152-L186).  This will likely require an additional build item (e.g. a `ConsoleFormatterBannerBuildItem` which contains a `RuntimeValue<Optional<Supplier<String>>>` or something of that sort) and a bit of core integration to wrap the default formatter with a delegating `ExtFormatter` which supplies the header and delegates the other methods to the default formatter.

The reason for this is:

* The banner should only appear on the console.  We don't want it in a file, published to syslog, ending up in fluentd, etc.
* The formatting of multi-line log messages is a bit awkward by default, with the first line indentation not matching up with the rest.
* If something like JSON formatting is enabled for the console, we don't want a banner because it won't end up making sense to whatever system is parsing it.

Reply:
So should I remove this recorder step and use only BannerProcessor and BannerConfig only ?

Reply:
I think the recorder should be modified to return the string supplier runtime value which in turn gets the banner as a string.  The processor step would pass that in to the logger setup using a build item, and the core logging setup processor would have to be modified to merge that in to the formatter configuration.